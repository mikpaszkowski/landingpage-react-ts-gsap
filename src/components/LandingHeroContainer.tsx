import * as React from "react";
import { CustomButton } from "./CustomButton";
import styled from "styled-components";
import { ReactComponent as Ellipse3 } from "../assets/svg-images/ellipse-3.svg";
import { ReactComponent as HeroImage } from "../assets/svg-images/hero-img.svg";
import { fontTypes, colors } from "../styles/styleConstants";

const LandingContainerWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-top: 15.3rem;
`;

const HeroHeadline = styled.div`
  font-size: 8rem;
  margin-left: 0.2rem;
  width: 42.3rem;
  line-height: 117%;
`;

const HeroMessage = styled.p`
  font-size: 2.4rem;
  line-height: 3.6rem;
  font-style: normal;
  font-weight: normal;
  font-family: ${fontTypes.secondaryFont};
  color: ${colors.secondaryFont};
  margin-bottom: 3.6rem;
  margin-top: 1.6rem;
  margin-left: 0.2rem;
  width: 42rem;
`;

const GetStartedBtn = styled(CustomButton)`
  padding: 1.6rem 1rem;
  width: 25.5rem;
`;

const HeroImageStyled = styled(HeroImage)`
  align-self: flex-end;
`;

const HeroLeftHalf = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  height: 43.4rem;
`;

export const LandingContainer = () => {
  return (
    <LandingContainerWrapper>
      <HeroLeftHalf>
        <HeroHeadline>
          Check y<Ellipse3 />
          ur well-being
        </HeroHeadline>
        <HeroMessage>
          We’re sure that deciding to check on your health is a big step, even
          if it doesn’t bother in any way.
        </HeroMessage>
        <GetStartedBtn fontsize="2.5rem" lineHeight="3.7rem">
          Get started
        </GetStartedBtn>
      </HeroLeftHalf>
      <HeroImageStyled />
    </LandingContainerWrapper>
  );
};
