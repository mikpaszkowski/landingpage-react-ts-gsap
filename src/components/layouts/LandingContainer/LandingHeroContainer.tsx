import * as React from "react";
import { CustomButton } from "../../ReusableComponents/CustomButton";
import styled from "styled-components";
import { ReactComponent as Ellipse3 } from "./ellipse-3.svg";
import { ReactComponent as HeroImage } from "./hero-img.svg";
import { fontTypes } from "../../../styles/constants";
import { device } from "../../../styles/responsive";

import {
  laptopLargeLeftSection,
  laptopLargeRightSection,
} from "../../../animations/landingHeroContainer";
import gsap from "gsap";
import { ScrollTrigger } from "gsap/dist/ScrollTrigger";
gsap.registerPlugin(ScrollTrigger);

const LandingContainerWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-top: 27.1rem;
  padding: 0 16.7rem;
  position: relative;
  width: 100%;

  @media ${device.laptopLarge} {
    flex-direction: row;
    padding: 0 16.7rem;
  }

  @media ${device.laptopSmall} {
    margin-top: 15rem;
    padding: 0 6rem;
  }

  @media ${device.tabletLarge} {
    flex-direction: column-reverse;
    padding: 0 2.4rem;
    margin-top: 12rem;
  }

  @media ${device.tablet} {
    padding: 0 2.4rem;
    margin-top: 12rem;
  }

  @media ${device.mobileMedium} {
    flex-direction: column-reverse;
    margin-top: 0;
    padding: 0 2.4rem;
  }
`;

const HeroHeadline = styled.h1`
  font-size: 8rem;
  margin-left: 0.2rem;
  line-height: 117%;
  font-style: normal;
  font-weight: normal;

  @media ${device.laptopLarge} {
    font-size: 8rem;
  }

  @media ${device.tabletLarge} {
    width: 100%;
    font-size: 7rem;
  }

  @media ${device.tablet} {
    font-size: 5rem;
    width: 100%;
  }

  @media ${device.mobileMedium} {
    font-size: 5rem;
    width: 100%;
  }
`;

const HeroMessage = styled.p`
  font-size: 2.4rem;
  line-height: 3.6rem;
  font-style: normal;
  font-weight: normal;
  font-family: ${fontTypes.secondaryFont};
  color: ${({ theme }) => theme.palette.secondary.font};
  margin-bottom: 3.6rem;
  margin-top: 1.6rem;
  margin-left: 0.2rem;

  @media ${device.laptopSmall} {
    font-size: 2.2rem;
    width: 100%;
  }

  @media ${device.tabletLarge} {
    width: 100%;
    font-size: 3rem;
  }

  @media ${device.mobileMedium} {
    text-align: left;
    font-size: 2rem;
    line-height: 3rem;
    width: 100%;
  }
`;

const GetStartedBtn = styled(CustomButton)`
  padding: 1.6rem 1rem;
  width: 25.5rem;
  margin-bottom: 1.6rem;

  @media ${device.tabletLarge} {
    width: 100%;
  }

  @media ${device.mobileMedium} {
    width: 100%;
  }
`;

const ImageWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 100%;
  height: 100%;
  align-self: flex-end;
  margin-top: 4.9rem;

  @media ${device.laptopLarge} {
    align-self: center;
  }

  @media ${device.tabletLarge} {
  }

  @media ${device.mobileMedium} {
    width: auto;
    margin-top: 2.4rem;
  }
`;

const HeroImageStyled = styled(HeroImage)`
  align-self: flex-end;
  width: 67.7rem;

  @media ${device.laptopSmall} {
    width: 55rem;
  }

  @media ${device.tabletLarge} {
    width: 90%;
    align-self: center;
    margin-bottom: 0;
  }

  @media ${device.mobileMedium} {
    min-width: auto;
    margin-bottom: 0;
  }
`;

const HeroLeftHalf = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;

  @media ${device.laptopSmall} {
    flex-direction: column;
    margin-top: 4.2rem;
  }

  @media ${device.mobileMedium} {
    margin-top: 4.2rem;
    flex-direction: column;
  }
`;

const Ellipse3Styled = styled(Ellipse3)`
  @media ${device.mobileMedium} {
    width: 2.7rem;
    height: 2rem;
  }
`;

export const LandingHeroContainer = () => {
  const { useRef, useEffect } = React;

  const leftScetionRef = useRef<HTMLDivElement>(null);
  const rightScetionRef = useRef<HTMLDivElement>(null);

  useEffect(() => {
    ScrollTrigger.matchMedia({
      "(min-width: 1440px)": function () {
        laptopLargeLeftSection(leftScetionRef);
        laptopLargeRightSection(rightScetionRef);
      },

      "(min-width: 500px)": function () {
        laptopLargeLeftSection(leftScetionRef);
        laptopLargeRightSection(rightScetionRef);
      },

      "(max-width: 499px)": function () {
        laptopLargeLeftSection(leftScetionRef);
        laptopLargeRightSection(rightScetionRef);
      },
    });
  });
  return (
    <LandingContainerWrapper>
      <HeroLeftHalf ref={leftScetionRef}>
        <HeroHeadline>
          Check y<Ellipse3Styled />
          ur well-being
        </HeroHeadline>
        <HeroMessage>
          We’re sure that deciding to check on your health is a big step, even
          if it doesn’t bother in any way.
        </HeroMessage>
        <GetStartedBtn
          fontSize="2.5rem"
          lineHeight="3.7rem"
          name="Get started"
        />
      </HeroLeftHalf>
      <ImageWrapper ref={rightScetionRef}>
        <HeroImageStyled />
      </ImageWrapper>
    </LandingContainerWrapper>
  );
};
