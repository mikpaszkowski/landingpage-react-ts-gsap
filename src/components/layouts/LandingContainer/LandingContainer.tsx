import * as React from "react";
import styled from "styled-components";
import { LandingHeroContainer } from "./LandingHeroContainer";
import { device } from "../../../styles/responsive";

const LandingPageWrapper = styled.div`
  position: relative;
  width: 100%;
  padding: 0 16.7rem;

  @media ${device.mobileMedium} {
    padding: 0 2.4rem;
  }
`;

export const LandingContainer = () => {
  return (
    <LandingPageWrapper>
      <LandingHeroContainer />
    </LandingPageWrapper>
  );
};
