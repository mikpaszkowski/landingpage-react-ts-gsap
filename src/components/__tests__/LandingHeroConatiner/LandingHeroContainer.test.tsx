import { cleanup } from "@testing-library/react";
import "jest-styled-components";
import { renderWithLightTheme, mountWithDarkTheme } from "../helpers";
import { LandingHeroContainer } from "../../layouts/LandingContainer/LandingHeroContainer";

afterEach(() => {
  cleanup();
});

window.matchMedia =
  window.matchMedia ||
  function () {
    return {
      matches: false,
      addListener: function () {},
      removeListener: function () {},
    };
  };

test("check render correctly <LandingHeroContainer /> component", () => {
  const tree = renderWithLightTheme(<LandingHeroContainer />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe("check the LandingHeroContainer - darkTheme", () => {
  const wrapper = mountWithDarkTheme(<LandingHeroContainer />);

  test("should contain two containers: div", () => {
    expect(wrapper.find("div").at(0).children().length).toEqual(2);
  });

  test("should contain one headline", () => {
    expect(wrapper.find(LandingHeroContainer).find("h1").length).toEqual(1);
  });

  describe("should contain p tag element", () => {
    const herotextMessage =
      "We’re sure that deciding to check on your health is a big step, even if it doesn’t bother in any way.";

    test("exactly one p tag element", () => {
      expect(wrapper.find(LandingHeroContainer).find("p").length).toEqual(1);
    });
    test("should contain text message inside", () => {
      expect(wrapper.find(LandingHeroContainer).find("p").text()).toEqual(
        herotextMessage
      );
    });
    test("should contain button", () => {
      expect(wrapper.find(LandingHeroContainer).find("button").text()).toEqual(
        "Get started"
      );
    });

    describe("should contain two svg images", () => {
      test("svg icon in headline", () => {
        expect(
          wrapper.find(LandingHeroContainer).find("svg").at(0).text()
        ).toEqual("ellipse-3.svg");
      });

      test("should contain svg image", () => {
        expect(
          wrapper.find(LandingHeroContainer).find("svg").at(1).text()
        ).toEqual("hero-img.svg");
      });
    });
  });
});
