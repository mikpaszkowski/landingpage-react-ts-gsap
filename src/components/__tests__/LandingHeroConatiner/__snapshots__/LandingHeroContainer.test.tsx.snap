// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check render correctly <LandingHeroContainer /> component 1`] = `
.c5 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.7rem 3.1rem;
  font-family: 'Reem Kufi',sans-serif;
  font-size: 2.5rem;
  font-style: normal;
  font-weight: normal;
  line-height: 3.7rem;
  color: #131E07;
  border: none;
  border-radius: 8px;
  background-color: #FABD20;
  cursor: pointer;
  -webkit-transition: all 0.25s ease-in-out;
  transition: all 0.25s ease-in-out;
  overflow: hidden;
  will-change: transform;
}

.c5::after {
  content: "";
  position: absolute;
  z-index: -1;
  bottom: -100%;
  left: 0;
  height: 100%;
  width: 100%;
  border-radius: 8px;
  background-color: black;
  color: #FABD20;
  -webkit-transition: all 0.15s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: all 0.15s cubic-bezier(0.455,0.03,0.515,0.955);
}

.c5:hover {
  box-shadow: 0px 1rem 2rem rgba(0,0,0,0.25);
  color: #FABD20;
}

.c5:hover::after {
  -webkit-transform: translateY(-100%);
  -ms-transform: translateY(-100%);
  transform: translateY(-100%);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 27.1rem;
  padding: 0 16.7rem;
  position: relative;
  width: 100%;
}

.c2 {
  font-size: 8rem;
  margin-left: 0.2rem;
  line-height: 117%;
  font-style: normal;
  font-weight: normal;
}

.c4 {
  font-size: 2.4rem;
  line-height: 3.6rem;
  font-style: normal;
  font-weight: normal;
  font-family: 'Reem Kufi',sans-serif;
  color: #40462F;
  margin-bottom: 3.6rem;
  margin-top: 1.6rem;
  margin-left: 0.2rem;
}

.c6 {
  padding: 1.6rem 1rem;
  width: 25.5rem;
  margin-bottom: 1.6rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  margin-top: 4.9rem;
}

.c8 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  width: 67.7rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

@media (max-width:1440px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    padding: 0 16.7rem;
  }
}

@media (max-width:1280px) {
  .c0 {
    margin-top: 15rem;
    padding: 0 6rem;
  }
}

@media (max-width:1110px) {
  .c0 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
    padding: 0 2.4rem;
    margin-top: 12rem;
  }
}

@media (max-width:768px) {
  .c0 {
    padding: 0 2.4rem;
    margin-top: 12rem;
  }
}

@media (max-width:375px) {
  .c0 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
    margin-top: 0;
    padding: 0 2.4rem;
  }
}

@media (max-width:1440px) {
  .c2 {
    font-size: 8rem;
  }
}

@media (max-width:1110px) {
  .c2 {
    width: 100%;
    font-size: 7rem;
  }
}

@media (max-width:768px) {
  .c2 {
    font-size: 5rem;
    width: 100%;
  }
}

@media (max-width:375px) {
  .c2 {
    font-size: 5rem;
    width: 100%;
  }
}

@media (max-width:1280px) {
  .c4 {
    font-size: 2.2rem;
    width: 100%;
  }
}

@media (max-width:1110px) {
  .c4 {
    width: 100%;
    font-size: 3rem;
  }
}

@media (max-width:375px) {
  .c4 {
    text-align: left;
    font-size: 2rem;
    line-height: 3rem;
    width: 100%;
  }
}

@media (max-width:1110px) {
  .c6 {
    width: 100%;
  }
}

@media (max-width:375px) {
  .c6 {
    width: 100%;
  }
}

@media (max-width:1440px) {
  .c7 {
    -webkit-align-self: center;
    -ms-flex-item-align: center;
    align-self: center;
  }
}

@media (max-width:1110px) {

}

@media (max-width:375px) {
  .c7 {
    width: auto;
    margin-top: 2.4rem;
  }
}

@media (max-width:1280px) {
  .c8 {
    width: 55rem;
  }
}

@media (max-width:1110px) {
  .c8 {
    width: 90%;
    -webkit-align-self: center;
    -ms-flex-item-align: center;
    align-self: center;
    margin-bottom: 0;
  }
}

@media (max-width:375px) {
  .c8 {
    min-width: auto;
    margin-bottom: 0;
  }
}

@media (max-width:1280px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    margin-top: 4.2rem;
  }
}

@media (max-width:375px) {
  .c1 {
    margin-top: 4.2rem;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:375px) {
  .c3 {
    width: 2.7rem;
    height: 2rem;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h1
      className="c2"
    >
      Check y
      <svg
        className="c3"
      >
        ellipse-3.svg
      </svg>
      ur well-being
    </h1>
    <p
      className="c4"
    >
      We’re sure that deciding to check on your health is a big step, even if it doesn’t bother in any way.
    </p>
    <button
      className="c5 c6"
      fontSize="2.5rem"
    >
      Get started
    </button>
  </div>
  <div
    className="c7"
  >
    <svg
      className="c8"
    >
      hero-img.svg
    </svg>
  </div>
</div>
`;
